examples/tv-app/android/java/ChannelManager.cpp

```diff
@@ -363,30 +371,273 @@ void ChannelManager::HandleGetProgramGuide(
+            needToFreeStrings.push_back(description);
+            if (jdescription != nullptr)
+            {
+                program->description = Optional<CharSpan>(description->charSpan());
+            }
+
+            jfieldID startTimeField = env->GetFieldID(programClass, "startTime", "J");
+            jlong jstartTime        = env->GetLongField(programObject, startTimeField);
+            program->startTime      = static_cast<uint32_t>(jstartTime);
+
+            jfieldID endTimeField = env->GetFieldID(programClass, "endTime", "J");
+            jlong jendTime        = env->GetLongField(programObject, endTimeField);
+            program->endTime      = static_cast<uint32_t>(jendTime);
+
+            uint32_t recordFlag               = 0;
+            jfieldID recordFlagScheduledField = env->GetFieldID(programClass, "recordFlagScheduled", "Z");
+            jboolean jrecordFlagScheduled     = env->GetBooleanField(programObject, recordFlagScheduledField);
+            if (jrecordFlagScheduled)
+            {
+                recordFlag |= 1;
+            }
+            jfieldID recordFlagSeriesField = env->GetFieldID(programClass, "recordFlagSeries", "Z");
+            jboolean jrecordFlagSeries     = env->GetBooleanField(programObject, recordFlagSeriesField);
+            if (jrecordFlagSeries)
+            {
+                recordFlag |= 2;
+            }
+            jfieldID recordFlagRecordedField = env->GetFieldID(programClass, "recordFlagRecorded", "Z");
+            jboolean jrecordFlagRecorded     = env->GetBooleanField(programObject, recordFlagRecordedField);
+            if (jrecordFlagRecorded)
+            {
+                recordFlag |= 4;
+            }
+
+            program->recordingFlag = MakeOptional(static_cast<uint32_t>(recordFlag));
+
+            needToFreePrograms.push_back(program);
+            tempPrograms.push_back(*program);
+        }
+
+        response.programList = DataModel::List<const ProgramType>(tempPrograms.data(), tempPrograms.size());
+
+        err = helper.Success(response);
+
+        for (ProgramType * program : needToFreePrograms)
+        {
+            delete program;
+        }
+
+        for (JniUtfString * str : needToFreeStrings)
+        {
+            delete str;
+        }
+    }
+
+exit:
+    ChipLogProgress(Zcl, " -- HandleGetProgramGuide -- 55");
+    if (err != CHIP_NO_ERROR)
+    {
+        ChipLogError(Zcl, "ChannelManager::HandleGetProgramGuide status error: %s", err.AsString());
+    }
 }
 
 bool ChannelManager::HandleRecordProgram(const chip::CharSpan & programIdentifier, bool shouldRecordSeries,
                                          const DataModel::DecodableList<AdditionalInfo> & externalIdList,
                                          const chip::ByteSpan & data)
 {
-    // Start recording
-    return true;
+    jboolean ret = JNI_FALSE;
+    JNIEnv * env = JniReferences::GetInstance().GetEnvForCurrentThread();
+
```

Comment:
Could you put the below JniLocalReferenceManager in the beginning of this HandleRecordProgram callback function so that we can automatically release local reference , for example, local reference generated from NewObjectArray, and CallBooleanMethod when it is out of scope.
JNIEnv * env = JniReferences::GetInstance().GetEnvForCurrentThread();
VerifyOrReturn(env != nullptr, ChipLogError(Controller, "Could not get JNIEnv for current thread"));
JniLocalReferenceManager manager(env);

Reply:
fixed - thanks!